core/src/main/scala/akka/kafka/internal/BaseSingleSourceLogic.scala

```diff
@@ -111,14 +111,14 @@ import scala.concurrent.{ExecutionContext, Future}
   setHandler(shape.out, new OutHandler {
     override def onPull(): Unit = pump()
     override def onDownstreamFinish(cause: Throwable): Unit =
-      performShutdown()
+      performShutdown(cause)
```

Comment:
We are not calling super.onDownstreamFinish by default?
I don't know if that is important? Could there be problems when just swallowing the signal as we do by default here?

Reply:
Good find. That should have been there.